iam/api-client/src/main/java/iam/snippets/Quickstart.java

```diff
@@ -13,48 +13,161 @@
+import com.google.api.services.cloudresourcemanager.model.SetIamPolicyRequest;
 import com.google.api.services.iam.v1.IamScopes;
-import com.google.api.services.iam.v1.model.ListRolesResponse;
-import com.google.api.services.iam.v1.model.Role;
 import com.google.auth.http.HttpCredentialsAdapter;
 import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
 import java.util.Collections;
 import java.util.List;
 
 public class Quickstart {
 
-  public static void main(String[] args) throws Exception {
-    // Get credentials
+  public static void main(String[] args) {
+    // TODO: Replace with your project ID.
+    String projectId = "your-project";
+    // TODO: Replace with the ID of your member in the form "user:member@example.com"
+    String member = "your-member";
+    // The role to be granted.
+    String role = "roles/logging.logWriter";
+
+    // Initializes the Cloud Resource Manager service.
+    CloudResourceManager crmService = null;
+    try {
+      crmService = initializeService();
+    } catch (IOException | GeneralSecurityException e) {
+      System.out.println("Unable to initialize service: \n" + e.toString());
+    }
+
+    // Grants your member the "Log writer" role for your project.
+    addBinding(crmService, projectId, member, role);
+
+    // Get the project's policy and print all members with the "Log Writer" role
+    Policy policy = getPolicy(crmService, projectId);
+    Binding binding = null;
+    List<Binding> bindings = policy.getBindings();
+    for (Binding b : bindings) {
+      if (b.getRole().equals(role)) {
+        binding = b;
+        break;
+      }
+    }
+    System.out.println("Role: " + binding.getRole());
+    System.out.print("Members: ");
+    for (String m : binding.getMembers()) {
+      System.out.print("[" + m + "] ");
+    }
+    System.out.println();
+
+    // Removes member from the "Log writer" role.
+    removeMember(crmService, projectId, member, role);
+  }
+
+  public static CloudResourceManager initializeService()
+      throws IOException, GeneralSecurityException {
+    // Use the Application Default Credentials strategy for authentication. For more info, see:
+    // https://cloud.google.com/docs/authentication/production#finding_credentials_automatically
     GoogleCredentials credential =
         GoogleCredentials.getApplicationDefault()
             .createScoped(Collections.singleton(IamScopes.CLOUD_PLATFORM));
 
-    // Create the Cloud IAM service object
-    Iam service =
-        new Iam.Builder(
+    // Creates the Cloud Resource Manager service object.
+    CloudResourceManager service =
+        new CloudResourceManager.Builder(
                 GoogleNetHttpTransport.newTrustedTransport(),
                 JacksonFactory.getDefaultInstance(),
                 new HttpCredentialsAdapter(credential))
-            .setApplicationName("quickstart")
+            .setApplicationName("service-accounts")
```

Comment:
It might be better if you used `"IAM Quickstart snippet"`

Reply:
Done.